src/form/Input.js

```diff
@@ -18,8 +18,10 @@
     Input.prototype._class += " form_Input";
     Input.prototype.implements(IInput.prototype);
 
-    Input.prototype.publish("type", "text", "set", "Input type", ["textbox", "number", "checkbox", "button", "select", "textarea", "date", "html-color"]);
+    Input.prototype.publish("type", "text", "set", "Input type", ["html-color", "number", "checkbox", "button", "select", "textarea", "date", "text", "range", "search", "email", "time", "datetime"]);
     Input.prototype.publish("selectOptions", [], "array", "Array of options used to fill a dropdown list");
+    Input.prototype.publish("low", null, "number", "Minimum value for Range input");
+    Input.prototype.publish("high", null, "number", "Maximum value for Range input");
```

Comment:
May as well add "step" to match the our Slider (or do we change slider to match this?) - since HTML5 supports it (see https://developer.mozilla.org/en/docs/Web/HTML/Element/Input)